girder_annotation/girder_large_image_annotation/rest/annotation.py

```diff
@@ -558,6 +561,94 @@ def deleteItemAnnotations(self, item):
+                             sortDir=False):
+        recursivePipeline = [
+            {'$graphLookup': {
+                'from': 'folder',
+                'startWith': '$_id',
+                'connectFromField': '_id',
+                'connectToField': 'parentId',
+                'as': '__children'
+            }},
+            {'$unwind': {'path': '$__children'}},
+            {'$replaceRoot': {'newRoot': '$__children'}}] if recurse else []
+        accessPipeline = [
+            {'$match': {
+                '$or': [
+                    {'access.users':
+                        {'$elemMatch': {
+                            'id': user['_id'],
+                            'level': {'$gte': 2}
+                        }}},
+                    {'access.groups':
+                        {'$elemMatch': {
+                            'id': {'$in': user['groups']},
+                            'level': {'$gte': 2}
+                        }}}
+                ]
+            }}
+        ] if not user['admin'] else []
+        pipeline = [
+            {'$match': {'_id': 'none'}},
+            {'$unionWith': {
+                'coll': 'folder',
+                'pipeline': [{'$match': {'_id': ObjectId(id)}}] +
+                recursivePipeline +
+                [{'$unionWith': {
+                    'coll': 'folder',
+                    'pipeline': [{'$match': {'_id': ObjectId(id)}}]
+                }}, {'$lookup': {
+                    'from': 'item',
+                    'localField': '_id',
+                    'foreignField': 'folderId',
+                    'as': '__items'
+                }}, {'$lookup': {
+                    'from': 'annotation',
+                    'localField': '__items._id',
+                    'foreignField': 'itemId',
+                    'as': '__annotations'
+                }}, {'$unwind': '$__annotations'},
+                    {'$replaceRoot': {'newRoot': '$__annotations'}},
+                    {'$match': {'_active': {'$ne': False}}}
+                ] + accessPipeline
+            }},
+        ]
+        pipeline = pipeline + [{'$sort': {sort: sortDir}}] if sort else pipeline
+        pipeline = pipeline + [{'$skip': offset}] if offset else pipeline
+        pipeline = pipeline + [{'$limit': limit}] if limit else pipeline
+
+        return Annotation().collection.aggregate(pipeline)
+
+    @autoDescribeRoute(
+        Description('Check if there are any annotations from the items in a folder')
+        .param('id', 'The ID of the folder', required=True, paramType='path')
+        .param('recurse', 'Whether or not to recursively check '
+               'subfolders for annotations', required=False, default=True, dataType='boolean')
+        .errorResponse()
+    )
+    @access.public
+    def existFolderAnnotations(self, id, recurse):
+        annotations = self.getFolderAnnotations(id, recurse, self.getCurrentUser(), 1)
+        try:
+            next(annotations)
+            yield True
```

Comment:
If you use `return` rather than `yield`, the endpoint will return `true` or `false` rather than `[true]` or `[false]`, which was more what I would have expected,

Reply:
That's a good point.

See 06302c7